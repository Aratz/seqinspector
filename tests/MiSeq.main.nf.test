nextflow_pipeline {

    name "Test Workflow main.nf on MiSeq data"
    script "../main.nf"
    tag "seqinspector"
    tag "PIPELINE"

    test("MiSeq data test") {

        when {
            params {
                outdir = "$outputDir"
                input  = "https://raw.githubusercontent.com/nf-core/test-datasets/seqinspector/testdata/MiSeq/samplesheet.csv"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path("$outputDir/multiqc/lanes/L1/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/lanes/L1/multiqc_data/multiqc_fastqc.txt"),
                    path("$outputDir/multiqc/lanes/L1/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/lanes/L1/multiqc_data/multiqc_software_versions.txt"),

                    path("$outputDir/multiqc/groups/P001/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/groups/P001/multiqc_data/multiqc_fastqc.txt"),
                    path("$outputDir/multiqc/groups/P001/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/groups/P001/multiqc_data/multiqc_software_versions.txt"),

                    path("$outputDir/multiqc/multiqc_data/multiqc_citations.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_fastqc.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt"),
                    path("$outputDir/multiqc/multiqc_data/multiqc_software_versions.txt"),
                    ).match()
                }
            )
        }
    }
}
