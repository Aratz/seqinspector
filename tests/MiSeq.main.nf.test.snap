{
    "MiSeq data test": {
        "content": [
            "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
            "multiqc_fastqc.txt:md5,692b8aed0614ed1655f2c1cbea1ba312",
            "multiqc_general_stats.txt:md5,630167d67d3f92408cd1a04422c7196f",
            "multiqc_software_versions.txt:md5,7452f1f7aae2a8a4066c2ef6cd5ceb95",
            "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
            "multiqc_fastqc.txt:md5,692b8aed0614ed1655f2c1cbea1ba312",
            "multiqc_general_stats.txt:md5,630167d67d3f92408cd1a04422c7196f",
            "multiqc_software_versions.txt:md5,7452f1f7aae2a8a4066c2ef6cd5ceb95",
            "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
            "multiqc_fastqc.txt:md5,692b8aed0614ed1655f2c1cbea1ba312",
            "multiqc_general_stats.txt:md5,630167d67d3f92408cd1a04422c7196f",
            "multiqc_software_versions.txt:md5,7452f1f7aae2a8a4066c2ef6cd5ceb95"
        ],
        "timestamp": "2024-05-08T16:29:33.284003"
    }
}